src/main/java/org/sundry/recipes/bootstrap/RecipeBootstrap.java

```diff
@@ -0,0 +1,200 @@
+  
+    if(!mexicanCatOptional.isPresent()) {
+      throw new RuntimeException("Expected Category not found");
+    }
+    
+    Category americanCategory = americanCatOptional.get();
+    Category mexicanCategory = mexicanCatOptional.get();
+  
+    Recipe guacRecipe = new Recipe();
+    guacRecipe.setDescription("Perfect Guacamole");
+    guacRecipe.setPrepTime(10);
+    guacRecipe.setCookTime(0);
+    guacRecipe.setDifficulty(Difficulty.EASY);
+    guacRecipe.setDirections("1 Cut avocado, remove flesh: Cut the avocados in half. Remove seed. Score the inside of the avocado with a blunt knife and scoop out the flesh with a spoon" +
+            "\n" +
+            "2 Mash with a fork: Using a fork, roughly mash the avocado. (Don't overdo it! The guacamole should be a little chunky.)" +
+            "\n" +
+            "3 Add salt, lime juice, and the rest: Sprinkle with salt and lime (or lemon) juice. The acid in the lime juice will provide some balance to the richness of the avocado and will help delay the avocados from turning brown.\n" +
+            "Add the chopped onion, cilantro, black pepper, and chiles. Chili peppers vary individually in their hotness. So, start with a half of one chili pepper and add to the guacamole to your desired degree of hotness.\n" +
+            "Remember that much of this is done to taste because of the variability in the fresh ingredients. Start with this recipe and adjust to your taste.\n" +
+            "4 Cover with plastic and chill to store: Place plastic wrap on the surface of the guacamole cover it and to prevent air reaching it. (The oxygen in the air causes oxidation which will turn the guacamole brown.) Refrigerate until ready to serve.\n" +
+            "Chilling tomatoes hurts their flavor, so if you want to add chopped tomato to your guacamole, add it just before serving.\n" +
+            "\n" +
+            "\n" +
+            "Read more: http://www.simplyrecipes.com/recipes/perfect_guacamole/#ixzz4jvpiV9Sd");
+  
+    Notes guacNotes = new Notes();
+  
+    guacNotes.setRecipeNotes("For a very quick guacamole just take a 1/4 cup of salsa and mix it in with your mashed avocados.\n" +
+            "Feel free to experiment! One classic Mexican guacamole has pomegranate seeds and chunks of peaches in it (a Diana Kennedy favorite). Try guacamole with added pineapple, mango, or strawberries.\n" +
+            "The simplest version of guacamole is just mashed avocados with salt. Don't let the lack of availability of other ingredients stop you from making guacamole.\n" +
+            "To extend a limited supply of avocados, add either sour cream or cottage cheese to your guacamole dip. Purists may be horrified, but so what? It tastes great.\n" +
+            "\n" +
+            "\n" +
+            "Read more: http://www.simplyrecipes.com/recipes/perfect_guacamole/#ixzz4jvoun5ws");
+    guacNotes.setRecipe(guacRecipe);
+    guacRecipe.setNotes(guacNotes);
```

Comment:
This creates a cyclical call when using equals to toString. When your using most ORMs only one object should own the relationship, in this case, Recipe should own Notes. So Recipe should have a Notes object, Notes should not have a Recipe reference.

Reply:
If Recipe owns Notes, how can I search for Recipe from Notes...?

Reply:
Good question. Many answers. In the RDMS/SQL world, you would assign an id to both and have a lookup table to map between them. This keeps your DTO clean, and most ORM frameworks support it with a few annotations. See Hibernate Lookup Table.

The real question is why do you want to look up a recipe from notes? The Recipe should be the parent and is the entry point, notes below to a recipe, but a recipe does not belong to notes.

Reply:
Ask your self this, would you ever be using notes with out the recipe?

Reply:
Made a new commit